packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts

```diff
@@ -615,3 +619,82 @@ async function insertAssociatedAccountsDataForProposal(
       .execute()
   );
 }
+
+async function calculateUserPerformance(onChainProposal: ProposalAccountWithKey) {
+    // calculate performance
+    const [ proposal ] = await usingDb(db => {
+      return db
+        .select()
+        .from(schema.proposals)
+        .where(eq(schema.proposals.proposalAcct, onChainProposal.publicKey.toString()))
+        .leftJoin(schema.daos, eq(schema.proposals.daoAcct, schema.daos.daoAcct))
+        .leftJoin(schema.tokens, eq(schema.daos.baseAcct, schema.tokens.mintAcct))
+        .limit(1)
+        .execute()
+    })
+
+    const { proposals, tokens } = proposal
+
+    const orders = await usingDb(db => {
+      return db
+        .select()
+        .from(schema.orders)
+        .where(inArray(schema.orders.marketAcct, [proposals.passMarketAcct, proposals.failMarketAcct]))
+        .execute()
+    })
+
+    let actors = <any>{}
+    
+    orders.forEach(o => {
+      const actor = o.actorAcct
+      if (!actors[actor]) {
+        actors[actor] = {
```

Comment:
Great stuff. Only thing to maybe consider is using reduce here instead of foreach. You can do Array.reduce and initialize with an empty object. That way less mutable data structures. Also the object for 'actors' can be typed using the actual type structure of your data and preferrably not just 'any'.

Reply:
May it is better to use `Map` here because this specific type is not used anywhere outside this context.

Reply:
Sure as long as you do 'Map<UserPerformance>' or something similar